packages/fxa-auth-server/lib/routes/subscriptions/paypal-notifications.ts

```diff
@@ -78,7 +83,19 @@ export class PayPalNotificationHandler extends PayPalHandler {
     }
   }
 
-  private async handleMpCancel(message: IpnMerchPmtType) {}
+  private async handleMpCancel(message: IpnMerchPmtType) {
+    const billingAgreement = await getPayPalBAByBAId(message.mp_id);
+    if (billingAgreement.status == 'Cancelled') {
+      return;
+    }
+    const accountCustomer = await getAccountCustomerByUid(billingAgreement.uid);
+    this.stripeHelper.removeCustomerPaypalAgreement(
+      accountCustomer.uid,
+      { id: accountCustomer.stripeCustomerId } as Stripe.Customer,
```

Comment:
If `removeCustomerPaypalAgreement` only needs the customerId, we should change that to use it instead of having to type-cast structures like this to pass them in.